add test and docs update for detached fork process

nanomosfet opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

This addresses issue #17592

vsemozhetbyt commented on doc/api/child_process.md:
```diff
@@ -325,6 +325,7 @@ changes:
 * `args` {string[]} List of string arguments.
 * `options` {Object}
   * `cwd` {string} Current working directory of the child process.
+  * `detached` {boolean} If `true`, Prepare child to run independently of its parent
```
Thank you!
Doc nits:
```suggestion
  * `detached` {boolean} If `true`, prepares child to run independently of its parent.
```

nanomosfet replied:
Will change it! Thanks for letting me know.

Also, ne za chto (Noticed your name)

lpinca commented on test/parallel/test-child-process-fork-detached.js:
```diff
@@ -0,0 +1,44 @@
+// Copyright Joyent, Inc. and other Node contributors.
```
AFAIK this copyright is not needed for new tests.

nanomosfet replied:
Will remove it. Thanks for letting me know.

vsemozhetbyt commented:
@nanomosfet I am sorry I did not notice this at first, but it seems we will have some linting issues with doc format and commit message format.

1. Long lines in docs are wrapped at 80 characters maximum.
2. At least the first commit's title and message need to conform with [this guideline](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines).

With these nits fixed, we will be ready to run CI tests.

nanomosfet commented:
@vsemozhetbyt Thanks for catching that.

 I updated the commit message. I couldn't tell if I should use test or docs for the sub system. 

I also changed the docs so that is is less than 80 characters with indent preserved.
Let me know what else I can do. :)

nanomosfet commented:
@vsemozhetbyt Looks like two of the checks failed. Not really sure what to make out of the details of the build. What is going on?

vsemozhetbyt commented:
1 Windows job failure may be unrelated. Let's rerun just this job:
https://ci.nodejs.org/job/node-test-commit-windows-fanned/22587/

nanomosfet commented:
@vsemozhetbyt Nice it looks like that one passed. Is that a flaky test?

Thank you BTW

vsemozhetbyt commented:
Flaky tests are usually marked as flaky in CI logs, but maybe we do not know yet that this one is flaky)

nanomosfet commented:
@vsemozhetbyt I see. What is left to do so we can merge?

I can change the anon. functions to arrow functions but I am afraid to because of rerunning the CI tests again. I can always make another PR to update a few test with arrow functions. I have noticed a lot of PR's where they are updating tests to use arrow functions.

vsemozhetbyt commented:
You can do this in another PR or just leave it for one of the Code-and-learn sessions (most of these latest PRs are from such session).

Our rules require at least 48 hours for the review process and 2 approvals. So, for now, we just need to wait a bit)

Trott commented on doc/api/child_process.md:
```diff
@@ -325,6 +325,8 @@ changes:
 * `args` {string[]} List of string arguments.
 * `options` {Object}
   * `cwd` {string} Current working directory of the child process.
+  * `detached` {boolean} If `true`, prepares child to run
```
Probably better to duplicate the text for the `detached` option under `spawn()` rather than have two different phrasings like this?

Trott commented on test/parallel/test-child-process-fork-detached.js:
```diff
@@ -0,0 +1,23 @@
+process.on('exit', function() {
+  assert.notStrictEqual(childId, -1);
+  // Killing the child process should not throw an error
+  process.kill(childId);
```
This strikes me as having significant potential for leaving stray processes around on CI. Not necessarily blocking, but surely something to keep an eye on after this lands....

Trott commented:
I would prefer either leaving the functions as they are, or else changing them here. Changing them to arrow functions in a separate PR is just churn for no good reason. Again, not blocking. Just my own preference.

vsemozhetbyt commented on test/parallel/test-child-process-fork-detached.js:
```diff
@@ -0,0 +1,23 @@
+const fork = require('child_process').fork;
+const fixtures = require('../common/fixtures');
+
+const nonPersistantNode = fork(
```
One more nit: `nonPersistantNode` -> `nonPersistentNode` here and below.

nanomosfet commented:
@codegagan Updated PR to use arrow functions 
@vsemozhetbyt fixed typo

> nit: nonPersistantNode -> nonPersistentNode here and below.

 @Trott - fork detached docs now match spwan detached. Both are now
```
 * `detached` {boolean} Prepare child to run independently of its parent
    process. Specific behavior depends on the platform, see
    [`options.detached`][]).
```
Re ran test and no stray node processes

Let me know what else I can do. Thank you for your time <3 :)

vsemozhetbyt commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/18931/

Trott commented:
Looks good to me. Can someone in @nodejs/child_process (or someone else knowledgable about potential pitfalls here) take a look?

Trott commented:
Landed in f051737ee4ebb7dfa25abc91f7a4797bd8fe0cb6.

Thanks for the contribution! 🎉

(If you're interested in other possible contributions to Node.js but don't have a good idea of where to start looking, some ideas are posted at https://www.nodetodo.org/next-steps/.)
